terraform/README.md

```diff
@@ -0,0 +1,117 @@
+    ├── dashboard/ 
+    ├── ecr/  
+    ├── alerts/
+    ├── pipeline/
+    ├── rds/
+    ├── reports/
+    ├── provider.tf
+    ├── variables.tf
+```
+
+### **Scripts**
+1. **`pipeline.tf`**
+    - Configure IAM roles and policies to grant permissions for Lambda and other AWS services.
+    - Define the Lambda function for the ETL pipeline and schedules it to run every 10 minutes.
+    - Manages the docker image for the Lambda function, including lifecycle policies to optimise storage.
+
+2. **`provider.tf`**
+    - Configures the AWS provider to use the `eu-west-2` region for resource deployment.
+    - Specifies the required AWS provider version for compatibility and stability.
+
+3. **`rds.tf`**
+    - Configures the ingress and egress rules to allow traffic on PostgreSQL port 5432.
+    - Provisions a PostgreSQL database with `db.t3.micro` instance type, automated backups, and performance insights enabled.
+    - Associates the database with the specified security group and makes it publicly available for connectivity.
+
+4. **`report.tf`**
+    - Creates an S3 bucket to store PDF reports with production environment tagging.
+    - Defines a lambda function for report generation, with IAM roles and policies granting access to S3, SES, RDS and CloudWatch.
+    - Schedules the lambda function to run daily and 9am via EventBridge.
+
+5. **`variables.tf`**
+    - Defines sensitive variables for database connection, including username, password, name, host, and port.
+    - Specifies the S3 bucket name for report storage and SES email addresses for sending and receiving reports.
+    - Ensures secure handling of sensitive data with defaults where applicable, like the database port.
+
+6. **`dashboard.tf`**
+    - Application Load Balancer (ALB) for routing traffic.
+    - ECS service for running the dashboard application.
+    - Security groups and IAM roles specific to the dashboard.
+
+7. **`alerts.tf`**
+    ...
+
+---
+
+## **How to Run the Terraform Scripts**
+
+### **Prerequisites**
+- Terraform 1.5 or later installed.
+- AWS CLI installed and configured with appropriate credentials.
+- The `.tfvars` file in the root directory containing the necessary variable values:
+    ```
+    db_user = "<Your RDS Username>"
+    db_password = "<Your RDS Password>"
+    db_name = "<Your Database Name>"
+    db_host = "<Your RDS Endpoint>"
+    s3_bucket_name = "<Your S3 Bucket Name>"
+    sender_email = "<Your SES Sender Email>"
+    recipient_email = "<Your SES Recipient Email>"
+    ```
+
+### **Provisioning ECRs and Configuring Components**
+Before proceeding with the deployment, ensure that Elastic Container Registries (ECRs) are provisioned, using the steps below. This step is necessary as each component’s deployment depends on pre-provisioned ECRs. Once the ECRs are ready, navigate to the respective subfolder within the Terraform directory and use the same process to configure and apply the deployment for each component individually.
+
```

Comment:
Nice! Sorry we might just need to add about adding the image url's to the files - I'm not sure if this is necessary but I guess it is a step in setting up so probably should